libs/execution_context/ebpf_native.c

```diff
@@ -42,18 +42,23 @@ typedef struct _ebpf_native_module_cleanup
     ebpf_preemptible_work_item_t* cleanup_workitem;
 } ebpf_native_module_cleanup_t;
 
+typedef enum _ebpf_native_module_state
+{
+    MODULE_STATE_UNINITIALIZED = 0,
+    MODULE_STATE_INITIALIZED,
+    MODULE_STATE_LOADING,
+    MODULE_STATE_LOADED,
+    MODULE_STATE_UNLOADING,
```

Comment:
We should use lower_snake_case for the enums.

https://github.com/microsoft/ebpf-for-windows/blob/b599ee24fd32582c659d33cbc46310b828fd5315/docs/DevelopmentGuide.md#naming-conventions

Reply:
Heads-up that at that link, it says (I didn't see any exception for enum types):

```
2. Use UPPER_SNAKE_CASE for macro names and constants.
```

Therefore, in mimicking how enum types are defined throughout the whole codebase, I implemented that way:

e.g. `ebpf_base.h`
```c
typedef enum _ebpf_return_type {
    EBPF_RETURN_TYPE_INTEGER = 0,
    EBPF_RETURN_TYPE_PTR_TO_MAP_VALUE_OR_NULL,
    EBPF_RETURN_TYPE_INTEGER_OR_NO_RETURN_IF_SUCCEED,
    EBPF_RETURN_TYPE_UNSUPPORTED,
} ebpf_return_type_t;
typedef enum _ebpf_argument_type {
...
}
```

Happy to change it to whatever best for this specific case though, please let me know.

Reply:
UPPER_SNAKE_CASE is correct for enum values

Reply:
It's fine for me, as that's exactly how I interpreted it.